src/components/chip/chip.ts

```diff
@@ -0,0 +1,129 @@
+import { html, LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+import { Constructor } from '../common/mixins/constructor.js';
+import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
+import { blazorTwoWayBind } from '../common/decorators';
+import { themes } from '../../theming';
+import { styles } from './themes/chip.base.css';
+import { styles as bootstrap } from './themes/chip.bootstrap.css';
+import { styles as fluent } from './themes/chip.fluent.css';
+import { styles as indigo } from './themes/chip.indigo.css';
+import { styles as material } from './themes/chip.material.css';
+
+export interface IgcChipEventMap {
+  igcRemove: CustomEvent<boolean>;
+  igcSelect: CustomEvent<boolean>;
+}
+/**
+ * Chips help people enter information, make selections, filter content, or trigger actions.
+ *
+ * @element igc-chip
+ *
+ * @slot - Renders the chip data.
+ * @slot prefix - Renders content before the data of the chip.
+ * @slot suffix - Renders content after the data of the chip.
+ *
+ * @fires igcRemove - Emits an event when the chip component is removed. Returns the removed chip component.
+ * @fires igcSelect - Emits event when the chip component is selected/deselected and any related animations and transitions also end.
+ *
+ * @csspart base - The base wrapper of the chip.
+ * @csspart prefix - The prefix container of the chip.
+ * @csspart suffix - The suffix container of the chip.
+ */
+@themes({ material, bootstrap, fluent, indigo })
+export default class IgcChipComponent extends EventEmitterMixin<
+  IgcChipEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-chip';
+
+  public static styles = styles;
+
+  /** Sets the disabled state for the chip. */
+  @property({ type: Boolean, reflect: true })
+  public disabled = false;
+
+  /** 	Defines if the chip is removable or not. */
+  @property({ type: Boolean, reflect: true })
+  public removable = false;
+
+  /**	Defines if the chip is selectable or not. */
+  @property({ type: Boolean, reflect: true })
+  public selectable = false;
+
+  /**	Defines if the chip is selected or not. */
+  @property({ type: Boolean, reflect: true })
+  @blazorTwoWayBind('igcSelected', 'detail')
+  public selected = false;
+
+  /** A property that sets the size of the chip component. */
+  @property({ reflect: true })
+  public size: 'small' | 'medium' | 'large' = 'medium';
+
+  /** A property that sets the color variant of the chip component. */
+  @property({ reflect: true })
+  public variant!:
```

Comment:
What should the default type be? Also, why do we need the `none` type. Shouldn't the default styles represent the `none` variant type?